### ISSUE TITLE
I find the model cann't repair the picture for example EnhanceCSI_example.png in the doc?

### ISSUE THREAD

**Issue by whaozl** [alexjc/neural-enhance#125]
```python
python3 enhance.py --zoom=1 --device=gpu --model=repair --type=photo /root/EnhanceCSI_example.png 
```

I picture this screenshot on the original EnhanceCSI_example.png.

![1](https://cloud.githubusercontent.com/assets/6327360/22625658/c01caf22-ebd7-11e6-8c6d-550a00a0a697.png)

using the cmd on the above, it has a successful output. the result is :
![1_ne1x](https://cloud.githubusercontent.com/assets/6327360/22625662/ce2478e8-ebd7-11e6-8a65-c6948adf526f.png)

the picture cann't deblur.

**Comment by zengxia** [comment]
yeah  I find the result the same as yours, the network seems do nothing to the image? Did i do something wrong? Have you got a better result?

**Comment by subzerofun** [comment]
The original image is already sharp – you can't see much blurring around the edges. So sharpening an already sharp image won't change that much. Maybe try it with a badly blurred photo?

Here are my results:

### **Original image 2048 x 1536 px:**
![crowd-original](https://cloud.githubusercontent.com/assets/17863119/24933828/c091381e-1f18-11e7-9c29-ae5ec5de7a2e.png)

---
### **Gaussian Blur 2px radius**
![crowd-gaussian-blur-2](https://cloud.githubusercontent.com/assets/17863119/24933852/e32d1dfc-1f18-11e7-87b2-908ea8666ad8.png)

### Result after 4 times neural enhance with --zoom=1 --model=deblur --type=photo:
![crowd-gaussian-blur-2_ne1x_ne1x_ne1x_ne1x](https://cloud.githubusercontent.com/assets/17863119/24933939/4c61654e-1f19-11e7-911c-8e981e8ea3a8.png)

---
### **Gaussian Blur 3px radius**
![crowd-gaussian-blur-3](https://cloud.githubusercontent.com/assets/17863119/24934007/9158cff2-1f19-11e7-92ed-ec0ca1904955.png)

### **Result 2x neural enhance deblur**
![crowd-gaussian-blur-3_ne1x_ne1x](https://cloud.githubusercontent.com/assets/17863119/24934014/99e899f4-1f19-11e7-9760-e44d78841b6a.png)

---

### **Gaussian Blur 5px radius**
![crowd-gaussian-blur-5](https://cloud.githubusercontent.com/assets/17863119/24934046/b3cf5952-1f19-11e7-85cf-c0e6712567bd.png)

### **Result 1x neural enhance deblur**
![crowd-gaussian-blur-5_ne1x](https://cloud.githubusercontent.com/assets/17863119/24934067/bf9966ce-1f19-11e7-8515-37b181932e53.png)

---

### **Field Blur (Photoshop) 5px**
![crowd-gaussian-fieldblur-5](https://cloud.githubusercontent.com/assets/17863119/24934088/dc65534e-1f19-11e7-8291-def572047ee9.png)

### **Result 3x neural enhance deblur**
![crowd-gaussian-fieldblur-5_ne1x_ne1x_ne1x](https://cloud.githubusercontent.com/assets/17863119/24934106/ff228f64-1f19-11e7-8eb5-b8644a0e1f26.png)

---

So for me it looks like the deblurring is working fine – but only for slightly blurred images.
It works quite OK for lens/field Blur. But it fails on motion blur.

Just take the original image and try it for yourself!

**Comment by subzerofun** [comment]
And here a few other results with a 5k (5184 x 3456 px) image:

### **Original:**

![crowd-5k-original](https://cloud.githubusercontent.com/assets/17863119/24934975/43c25924-1f1f-11e7-8642-a75122e11eac.jpg)

### **3 px Blur:**
![crowd-5k-gaussianblur-3px](https://cloud.githubusercontent.com/assets/17863119/24934983/579fc2e2-1f1f-11e7-9027-8d458fa7c0c9.jpg)


### **Result after 3x neural enhance:**
![crowd-5k-gaussianblur-3px_ne1x_ne1x_ne1x](https://cloud.githubusercontent.com/assets/17863119/24934998/6a802c1c-1f1f-11e7-9705-ef50bada2514.jpg)

---

### **5 px Blur:**
![crowd-5k-gaussianblur-5px](https://cloud.githubusercontent.com/assets/17863119/24935044/a07343a4-1f1f-11e7-91b1-c9ccbdb590c0.jpg)


### **Result after 3x neural enhance:**
![crowd-5k-gaussianblur-5px_ne1x_ne1x_ne1x](https://cloud.githubusercontent.com/assets/17863119/24935047/a7aa94a6-1f1f-11e7-8c26-00ef830caf87.jpg)

**Comment by zengxia** [comment]
Good job!
Thank you so much! May be i misunderstand the ability of this network, i used to think it is able to remove mosaic as the google paper "Pixel Recursive Super Resolution" says, at least something like that. Have you tried that?

**Comment by subzerofun** [comment]
But the Readme clearly states that the main purpose of the project is to upscale images – the name "neural _enhance_ " is a reference to all the cringeworthy scenes of TV crime series where they use special software to enhance/deblur/enlarge images in a way that's impossible in reality. Sorry for explaining this again – i hope this doesn't sound condescending (not my intention!), i just thought you haven't read it.

So in my opinion the only thing it has in common with the [Pixel Recursive Super Resolution ](https://arxiv.org/pdf/1702.00783.pdf) paper is that they both use a GAN. The PRSS solves a different kind of problem – the images mentioned in the paper only have a resolution of 8 x 8 px. So the generator has to hallucinate much more details than the one from neural enhance. There is no high frequency information available in the 8 x 8 px, so the PRSS cannot recreate the original image. It's only goal is to recreate a _plausible_ image.
If the output would look a lot like the original image, then it's certainly just coincidence – or that this particular face appeared frequently in the training images.

Neural enhance on the other hand can actually recover details in a much more practical way – you can see in my examples that the recreated high frequency contrast is not far from the sharp edges of the original images. 

Disclaimer: i'm no an expert on this field – i'm not even a ML student! So i hope i haven't written something that's completely wrong 😬.

**Comment by okfirelee** [comment]
@subzerofun  why yours img in docker program it not result file  and no error .
https://cloud.githubusercontent.com/assets/17863119/24934983/579fc2e2-1f1f-11e7-9027-8d458fa7c0c9.jpg
it s too big ?

**Comment by zengxia** [comment]
Hi,
   Sorry to reply late for my preparing some interview these days .
First, Thank you so much for your patience!  I'm so appreciate that you
explain these details for me! Actually, this is not my main research area
either, just want to follow some advanced   technology, maybe apply it to
my project if it is cool .
   Thank you again!
  Xia Zeng

2017-04-15 20:58 GMT+08:00 Firelee <notifications@github.com>:

**Comment by whaozl** [comment]
Thank you very much! This is a very nice work!
